fix(SingleEntitySelectMenuItems): extract `Add New` button from entitiesToSelect

nicolasrouanne opened the pull request:
# Description
Closes https://github.com/twentyhq/twenty/issues/8169

Extract `Add New` button from entitiesToSelect and add it as a separate element .
There doesn't seem to be a point in having `Add New` as part of a list, it seems better off in its own component, apart from list items

nicolasrouanne committed:
fix(SingleEntitySelectMenuItems): extract `Add New` button from entitiesToSelect

- and add it as a separate element
- to be consistent with `MultiRecordSelect`

nicolasrouanne commented on packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx:
```diff
@@ -104,6 +96,8 @@ export const SingleEntitySelectMenuItemsWithSearch = ({
         </>
       )}
+      {entities.entitiesToSelect.length > 0 && <DropdownMenuSeparator />}
+      {createNewButton}
```
I didn't wrap the `createNewButton` in a `<DropdownMenuItemsContainer>` although it's done in `MultiRecordSelect`
https://github.com/nicolasrouanne/twenty/blob/58cb1f769aed1053899e98a7599b47cde580fe0e/packages/twenty-front/src/modules/object-record/relation-picker/components/MultiRecordSelect.tsx#L178-L180

I don't understand the use of wrapping a simple button in this container, but maybe I'm missing something

FelixMalfait replied:
I haven't checked out the PR to look at the result yet but I'm guess this might have an impact on styling, probably the padding?
<img width="452" alt="Screenshot 2024-11-12 at 20 38 30" src="https://github.com/user-attachments/assets/fefea0a6-3af0-4895-9975-35f001efbdf9">

nicolasrouanne replied:
Hum I might have overlooked that. Will definitely check. Thanks!

nicolasrouanne commented on packages/twenty-front/src/modules/object-record/relation-picker/components/SingleEntitySelectMenuItemsWithSearch.tsx:
While working on this file I wondered why we have 2 naming conventions: `MultiRecordSelect` vs `SingleEntitySelect...`. Is there a reason?
If now, we could maybe use the same wording, either `Record` or `Entity`?

FelixMalfait replied:
I agree there's something wrong with naming, at first I thought it might be because it handles case like Select/Multi-Select fields because it's not records but it's not the case... 
Also there is "RelationPicker" which doesn't make much sense as it's only for single entity.
Feel free to rename with what you feel would be best! Thanks a lot!

nicolasrouanne committed:
fix(SingleEntitySelectMenuItems): wrap "Add New" in <DropdownMenuItemsContainer /> for padding

nicolasrouanne committed:
fix(SingleEntitySelectMenuItems): remove deleted prop from type

nicolasrouanne committed:
fix(SingleEntitySelectMenuItems): correctly position "Add New" in case of an "*end" positioned `dropdownPlacement`

- display is inverted when the `Placement` is at the `end`, following the `MultiRecordSelect` component design
- add a story to display the `end` placement
- pass down the `dropdownPlacement` prop in `SingleEntitySelect`

nicolasrouanne committed:
fix(SingleEntitySelect): revert exposing `dropdownPlacement` prop in `SingleEntitySelect`

- and passing it down to `SingleEntitySelectMenuItemsWithSearch`
- since it broke the position of the dropdown, see https://github.com/twentyhq/twenty/pull/8474#issuecomment-2474405877

FelixMalfait committed:
Don't shop separator if no create

FelixMalfait committed:
Merge main

FelixMalfait committed:
Small ui adjustments
